File: Remove Nth Node From End of List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode *temp1 = head;
        ListNode *temp2 = temp1;
        int l = Length(temp1);
        int a = l-n;
        int b=0;
        if(a==0){
            head = temp1->next;
            temp1 = NULL;
            return head;
        }
        while(temp1!=NULL && b<a){
            temp2 = temp1;
            temp1 = temp1->next;
            b+=1;
        }
        temp2->next = temp1->next;
        temp1 = NULL;
        return head;
        
    }
    
    int Length(ListNode *head){
        ListNode *p = head;
        int l = 0;
        while(p!=NULL){
            p = p->next;
            l+=1;
        }
        return l;
    }
};

File: Remove Nth Node From End of List.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: ListNode
        :type n: int
        :rtype: ListNode
        """
        # fast and slow pointer and prev
        # move fast n steps until end of list
        # return slow
        temp = head
        slow = temp
        fast = temp
        prev = slow
        while fast and n>0:
            fast = fast.next
            n -= 1
        while fast:
            fast = fast.next
            prev = slow
            slow = slow.next
        prev.next = slow.next
        if slow==head:
            return head.next
        return head
        